src/foam/nanos/auth/AuthorizationException.java

```diff
@@ -5,15 +5,32 @@
  */
 
 package foam.nanos.auth;
+import foam.box.CapabilityRequiredRemoteException;
 
 /**
  * Throw this when a user is logged in, but is not permitted to do something.
  */
 public class AuthorizationException extends SecurityException {
+  // TODO: This should be decoupled from AuthorizationException after
+  //       the modelled exception feature is merged.
+  private CapabilityRequiredRemoteException capabilityRequired_;
```

Comment:
I was thinking of having the server box set capabilityRequired_ if applicable (based on if the permission is intercepted by any capabilities) so that someone throwing an authorization exception doesn't have to worry about capabilities - what do you think of this solution?

Reply:
Yeah, that's what I meant by

> Then in the box code you could check if the exception has a permission set and if so, look up the set of capabilities that satisfy that permission and attach a list of those ids to the message.

but I guess it was clearer in my head, haha.

Reply:
Oh, I actually mis-remembered the way I did something. Yeah I'll for sure have to remove that property from AuthorizationException and put one for permission there instead